### PR TITLE
Feature/new sub directory  from  metakey annotation_source

### PR DESCRIPTION
**Pull request by vinay-ebi**
Fixes ENSPROD-8309

### CODE REVIEW

**Comment by vinay-ebi** [comment]
> @vinay-ebi This was quick! but I am afraid we don't want the sub dir only for geneset, but for all dirs. I just realised sometime, we'll have a different `genome` because ... some may contains mitochondrial, and some may not. Sorry!

Will this be the same meta-key species.annotation_source for genome/RNAseq/variation

**Comment by marcoooo** [comment]
> > @vinay-ebi This was quick! but I am afraid we don't want the sub dir only for geneset, but for all dirs. I just realised sometime, we'll have a different `genome` because ... some may contains mitochondrial, and some may not. Sorry!
> 
> Will this be the same meta-key species.annotation_source for genome/RNAseq/variation

yes for all directories.

File: modules/Bio/EnsEMBL/Production/Pipeline/FileDump/DirectoryPaths.pm [hunk 1]
```diff
@@ -64,7 +65,8 @@ sub write_output {
     output_dir      => $self->param('output_dir'),
     timestamped_dir => $self->param('timestamped_dir'),
     web_dir         => $self->param('web_dir'),
-    ftp_dir         => $self->param('ftp_dir')
+    ftp_dir         => $self->param('ftp_dir'),
+    annotation_source => $self->param_required('annotation_source')
```
[the remark this answers is not in the document]
Comment (vinay-ebi) [reply modules/Bio/EnsEMBL/Production/Pipeline/FileDump/DirectoryPaths.pm:69]: Ya, it can be simple param, but  just to make this parameter mandatory

**Comment by marcoooo** [comment]
@vinay-ebi please can we change the target branch to 108?

File: modules/Bio/EnsEMBL/Production/Pipeline/FileDump/DirectoryPaths.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (vinay-ebi) [reply modules/Bio/EnsEMBL/Production/Pipeline/FileDump/DirectoryPaths.pm:69]: ya default set to ensembl https://github.com/Ensembl/ensembl-production/blob/feature/new_sub_dir_annt_source/modules/Bio/EnsEMBL/Production/Pipeline/FileDump/DirectoryPaths.pm#L41

**Comment by vinay-ebi** [comment]
> @vinay-ebi please can we change the target branch to 108?

Other PR for 107/ 108 are ready I thought once this is approved I will raise those PRs

**Pull request merged**